### PR TITLE
VideoPlayer: WinRenderer - add smooth motion

### PR DESCRIPTION
**Pull request by afedchin**
see title

a lot of changes because of reformatting

### CODE REVIEW

**Comment by garbear** [comment]
feel like separating the changes? I'd like to learn a little more how this works and it's easier to digest when the reformatting is separate.

**Comment by jjd-uk** [comment]
@garbear in case you're not aware, this feature has already been in for OpenGL for a while https://github.com/xbmc/xbmc/pull/12805

**Comment by popcornmix** [comment]
I much prefer cosmetic and functional changes to be in separate commits.
It's almost impossible to tell where the functional changes are here.
And if somewhere down the line you wanted to revert this when troubleshooting the cosmetics will likely give you a lot more conflicts than the functional changes.

**Comment by afedchin** [comment]
> I much prefer cosmetic and functional changes to be in separate commits.

I agree, but VS does this automatically. I'll try to separate changes.

File: xbmc/cores/VideoPlayer/VideoRenderers/WinRenderer.cpp [hunk 1]
```diff
@@ -62,31 +63,32 @@ bool CWinRenderer::Register()
...
+    , m_renderMethod(RENDER_PS)
+    , m_bufferFormat(BUFFER_FMT_NONE)
+    , m_scalingMethod(VS_SCALINGMETHOD_LINEAR)
+    , m_scalingMethodGui(static_cast<ESCALINGMETHOD>(-1))
+    , m_processor(nullptr)
+    , m_sw_scale_ctx(nullptr)
+    , m_colorShader(nullptr)
+    , m_scalerShader(nullptr)
+    , m_capture(nullptr)
+    , m_pCLUTView(nullptr)
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/VideoRenderers/WinRenderer.cpp]: use default member initializers

**Comment by afedchin** [comment]
reformatting dropped.

**Comment by fritsch** [comment]
Some windows users say that this breaks their deinterlacing: https://forum.kodi.tv/showthread.php?tid=326616&pid=2764089#pid2764089 - investigation ongoing as test results seem a bit "bogus"(TM) for now.

**Comment by afedchin** [comment]
> I guess the issue is related to the fact that win does deinterlacing in renderer
yes, reverting this PR doesn't solves the issue.

**Comment by Andy2244** [comment]
@afedchin @thexai Can we get some form of "smooth motion" back for windows somehow?

My main problem is, that i miss a MadVR "smooth motion" or mpv "interpolation" type feature in Kodi. This PR was reverted in 2018 and reading the issues, it seems the revert did not actually solve the reported problems?

So maybe in 2020 this feature might be picked-up again for the new 19.x branch?

**Comment by Sebastiii** [comment]
Do we have some news about adding smooth motion to kodi ?
Thanks

**Pull request merged**